index.js

```diff
@@ -47,25 +48,47 @@ module.exports = function(source, sourceMap) {
                 // apply filename placeholders
                 baggageFile = util.applyPlaceholders(baggageFile, srcDirname, srcFilename);
 
-                // check for flag indicating this inclusion is optional
-                if (baggageVar.length && baggageVar.indexOf(optionalFlag) !== -1) {
+                // check for flags indicating this inclusion is optional
+                if (baggageVar.length) {
+                  // optional variable that can be ignored
+                  if (baggageVar.indexOf(optionalFlag) !== -1) {
                     ignoreArr = baggageVar.split(optionalFlag);
                     // term before the optional flag is the term that indicates, when present in a file, 
                     // that the require should be ignored for that file
                     if (ignoreArr[0]) {
                         ignoreFlag = ignoreArr[0];
                     }
                     baggageVar = ignoreArr[1];
+                  }
+                  
+                  // optional variable that can be added if requested
+                  if (baggageVar.indexOf(optionalAddFlag) !== -1) {
+                    addArr = baggageVar.split(optionalAddFlag);
+                    if (addArr[0]) {
+                      addFlag = addArr[0];
+                    }
+                    baggageVar = addArr[1];
+                  }
+                }
+                
+                // Check wether or not we should add the require, based on optional params
+                var includeRequire = true;
+                if (addArr) {
```

Comment:
Can you skip the check on `addArr` and just check for `addFlag`? `addArr` doesn't tell you anything definitive here